cli/src/config.rs

```diff
@@ -249,12 +252,18 @@ struct ConfigEnv {
 }
 
 impl ConfigEnv {
-    fn new() -> Self {
-        ConfigEnv {
-            config_dir: dirs::config_dir(),
-            home_dir: dirs::home_dir(),
+    fn new() -> Result<Self, ConfigError> {
+        // get the current platform's config dir. On MacOS that's actually the data_dir
+        #[cfg(not(target_os = "macos"))]
+        let platform_config_dir = Some(etcetera::choose_base_strategy()?.config_dir());
+        #[cfg(target_os = "macos")]
+        let platform_config_dir = Some(etcetera::base_strategy::Apple::new()?.data_dir());
+
+        Ok(ConfigEnv {
+            config_dir: platform_config_dir,
+            home_dir: etcetera::home_dir().ok(),
             jj_config: env::var("JJ_CONFIG").ok(),
-        }
+        })
```

Comment:
The problem is that if one fails, both will, because they both hinge on finding the home directory. That's true for the existing code, so IIUC there was never a situation where it picked $HOME over platform dir, and remains true for this one. I'd prefer to know about that (unlikely as it is, since it doesn't just look for $HOME but also uses some system calls to figure it out).
But if you prefer I can just return None.

Reply:
> The problem is that if one fails, both will, because they both hinge on finding the home directory.

Is that true if `$XDG_CONFIG_HOME` points outside of `$HOME`? Anyway, I think you're right that it's better to error out if we can't find the config directory. Assuming that's what we decide, shouldn't that be true whether we use `dirs` or `etcetera`? I.e., I don't think we should combine the change in behavior with the change from `dirs` to `etcetera`. Does that make sense?

Reply:
> Is that true if `$XDG_CONFIG_HOME` points outside of `$HOME`? 

You're right, in that instance it isn't true.

> Anyway, I think you're right that it's better to error out if we can't find the config directory. Assuming that's what we decide, shouldn't that be true whether we use `dirs` or `etcetera`? I.e., I don't think we should combine the change in behavior with the change from `dirs` to `etcetera`. Does that make sense?

I think you're saying that we should have etcetera not error, but convert to `None` instead, and continue having `ConfigEnv::new` return just `Self` instead of a `Result`? Then if we want to have it error instead, we can do that later?

Reply:
> I think you're saying that we should have etcetera not error, but convert to `None` instead, and continue having `ConfigEnv::new` return just `Self` instead of a `Result`? Then if we want to have it error instead, we can do that later?

Yes, that seems like the safest bet (least controversial). The other option is to instead start by changing the behavior while still using `dirs` in a first commit (or several if it gets complicated), and then switch to `etcetera` in a later commit. The risk with that approach is that people will object and it ends up blocking the later changes you want to make (adding support for `XDG_CONFIG_HOME` on MacOS).

Reply:
Yeah, and `dirs` doesn't return errors anyway, so not sure how to handle that (turn None into an Err??). I'll make the change.

Reply:
Alright, done.